astroid/brain/brain_re.py

```diff
@@ -34,6 +35,30 @@ def _re_transform():
     TEMPLATE = sre_compile.SRE_FLAG_TEMPLATE
     T = TEMPLATE
     DEBUG = sre_compile.SRE_FLAG_DEBUG
+    """
+        )
+    # On 3.11+ all flags come from re._compiler
+    return parse(
+        """
+    import re._compiler as _compiler
+    NOFLAG = 0
+    ASCII = _compiler.SRE_FLAG_ASCII
+    IGNORECASE = _compiler.SRE_FLAG_IGNORECASE
```

Comment:
Is there a way to do this without reiterating the whole thing? Like
```python
if PY311_PLUS:
    import_compiler = "import re._compiler as _compiler"
else:
    import_compiler = "import sre_compiler as _compiler"
return parse(f"""
{import_compiler}
ASCII = _compiler.SRE_FLAG_ASCII
...
```
...unless there a functional impact from doing so?

Reply:
Probably. Let me have a look!